src/graph/executor/algo/IsomorExecutor.cpp

```diff
@@ -12,77 +12,61 @@ namespace graph {
+   auto iterDE = ectx_->getResult(isomor->getdScanEOut()).iter();
+   auto iterQE = ectx_->getResult(isomor->getqScanEOut()).iter();
 
-   auto iter = ectx_->getResult(subgraph->inputVar()).iter();
-   auto gnSize = iter->size();
+   unsigned int v_count = iterDV->size();
+   unsigned int l_count = iterDV->size();
+   unsigned int e_count = iterDE->size();
 
-   ResultBuilder builder;
-   builder.value(iter->valuePtr());
+   unsigned int *offset = new unsigned int[v_count + 2];
+   unsigned int *neighbors = new unsigned int[e_count * 2];
+   unsigned int *labels = new unsigned int[l_count];
+   // load data vertices id and tags
+   while (iterDV->valid()) {
+     const auto v = iterDV->getColumn(nebula::kVid);  // check if v is a vertex
+     auto v_id = v.getInt();
+     const auto v2 = iterDV->getColumn(1);  // get label
+     auto l_id = v2.getInt();
+     // unsigned int v_id = (unsigned int)v.getInt(0);
+     labels[v_id] = l_id;  // Tag Id
+     iterDV->next();
+   }
 
-   std::unordered_map<Value, int64_t> currentVids;
-   currentVids.reserve(gnSize);
-   historyVids_.reserve(historyVids_.size() + gnSize);
-   if (currentStep == 1) {
-     for (; iter->valid(); iter->next()) {
-       const auto& src = iter->getColumn(nebula::kVid);
-       currentVids.emplace(src, 0);
-     }
-     iter->reset();
+   // load edges degree
+   while (iterDE->valid()) {
+     auto s = iterDE->getEdgeProp("*", kSrc);
+     unsigned int src = s.getInt();
+     offset[src]++;
+     iterDE->next();
    }
-   auto& biDirectEdgeTypes = subgraph->biDirectEdgeTypes();
-   while (iter->valid()) {
-     const auto& dst = iter->getEdgeProp("*", nebula::kDst);
-     auto findIter = historyVids_.find(dst);
-     if (findIter != historyVids_.end()) {
-       if (biDirectEdgeTypes.empty()) {
-         iter->next();
-       } else {
-         const auto& typeVal = iter->getEdgeProp("*", nebula::kType);
-         if (UNLIKELY(!typeVal.isInt())) {
-           iter->erase();
-           continue;
-         }
-         auto type = typeVal.getInt();
-         if (biDirectEdgeTypes.find(type) != biDirectEdgeTypes.end()) {
-           if (type < 0 || findIter->second + 2 == currentStep) {
-             iter->erase();
-           } else {
-             iter->next();
-           }
-         } else {
-           iter->next();
-         }
-       }
-     } else {
-       if (currentStep == steps) {
-         iter->erase();
-         continue;
-       }
-       if (currentVids.emplace(dst, currentStep).second) {
-         Row row;
-         row.values.emplace_back(std::move(dst));
-         ds.rows.emplace_back(std::move(row));
-       }
-       iter->next();
-     }
+
+   // load data edges
+   offset[0] = 0;
+   iterDE = ectx_->getResult(isomor->getdScanEOut()).iter();
+   while (iterDE->valid()) {
+     unsigned int src = iterDE->getEdgeProp("*", kSrc).getInt();
+     unsigned int dst = iterDE->getEdgeProp("*", kDst).getInt();
+
+     neighbors[offset[src + 1]] = dst;
```

Comment:
Do not understand this. Please explain via comment.

Reply:
(1) src and dst is the source and destination information.
(2) offset[src + 1] will be the position of the array.

Reply:
I can change it and comment it in the new version